From a93b8ec44817f2dd3d318cea8079663ca56bc238 Mon Sep 17 00:00:00 2001
From: Alexander Stein <alexander.stein@ew.tq-group.com>
Date: Thu, 10 Apr 2025 13:01:19 +0200
Subject: [PATCH 7/7] bat: Fix buffer time configuration

By swapping time configuration some hardware constraints can be supported.
Fixes the error: Set parameter to device error: buffer time: 500000
hw:tqmtlv320aic32: Invalid argument(-22)

Signed-off-by: Alexander Stein <alexander.stein@ew.tq-group.com>
Signed-off-by: Takashi Iwai <tiwai@suse.de>
(cherry picked from commit 7cf3bd535e4c4bfe11e085ae7e685f931b296db1)
Signed-off-by: Markus Niebel <Markus.Niebel@ew.tq-group.com>
---
 bat/alsa.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/bat/alsa.c b/bat/alsa.c
index 1559ed4..5066535 100644
--- a/bat/alsa.c
+++ b/bat/alsa.c
@@ -220,23 +220,23 @@ static int set_snd_pcm_params(struct bat *bat, struct pcm_container *sndpcm)
 
 		period_time = buffer_time / DIV_BUFFERTIME;
 
-		/* Set buffer time and period time */
-		err = snd_pcm_hw_params_set_buffer_time_near(sndpcm->handle,
-				params, &buffer_time, 0);
+		/* Set period time and buffer time */
+		err = snd_pcm_hw_params_set_period_time_near(sndpcm->handle,
+			params, &period_time, 0);
 		if (err < 0) {
 			fprintf(bat->err, _("Set parameter to device error: "));
-			fprintf(bat->err, _("buffer time: %d %s: %s(%d)\n"),
-					buffer_time,
+			fprintf(bat->err, _("period time: %d %s: %s(%d)\n"),
+					period_time,
 					device_name, snd_strerror(err), err);
 			return err;
 		}
 
-		err = snd_pcm_hw_params_set_period_time_near(sndpcm->handle,
-				params, &period_time, 0);
+		err = snd_pcm_hw_params_set_buffer_time_near(sndpcm->handle,
+				params, &buffer_time, 0);
 		if (err < 0) {
 			fprintf(bat->err, _("Set parameter to device error: "));
-			fprintf(bat->err, _("period time: %d %s: %s(%d)\n"),
-					period_time,
+			fprintf(bat->err, _("buffer time: %d %s: %s(%d)\n"),
+					buffer_time,
 					device_name, snd_strerror(err), err);
 			return err;
 		}
-- 
2.43.0

